Printing test report with failed chained test errors

Came across this nice one recently.

MWE:
```Julia
using Test
using TestReports

ts = @testset ReportingTestSet "test" begin
    @test true && false
end

report(ts)
```

Results in this error:

```
ERROR: AssertionError: isempty(XML_GLOBAL_ERROR_STACK)
Stacktrace:
 [1] macro expansion at /Users/malmiller/.julia/packages/EzXML/BF7Lf/src/error.jl:49 [inlined]
 [2] ElementNode(::String) at /Users/malmiller/.julia/packages/EzXML/BF7Lf/src/node.jl:479
 [3] failure_xml(::String, ::String, ::String) at /Users/malmiller/repos/TestReports.jl/src/to_xml.jl:43
 [4] to_xml(::Test.Fail) at /Users/malmiller/repos/TestReports.jl/src/to_xml.jl:120
 [5] (::TestReports.var"#4#5")(::Test.Fail) at /Users/malmiller/repos/TestReports.jl/src/to_xml.jl:102
 [6] iterate at ./generator.jl:47 [inlined]
 [7] _collect(::Array{Test.Fail,1}, ::Base.Generator{Array{Test.Fail,1},TestReports.var"#4#5"}, ::Base.EltypeUnknown, ::Base.HasShape{1}) at ./array.jl:699
 [8] collect_similar(::Array{Test.Fail,1}, ::Base.Generator{Array{Test.Fail,1},TestReports.var"#4#5"}) at ./array.jl:628
 [9] map(::Function, ::Array{Test.Fail,1}) at ./abstractarray.jl:2162
 [10] to_xml(::ReportingTestSet) at /Users/malmiller/repos/TestReports.jl/src/to_xml.jl:101
 [11] (::TestReports.var"#2#3")(::ReportingTestSet) at /Users/malmiller/repos/TestReports.jl/src/to_xml.jl:78
 [12] iterate at ./generator.jl:47 [inlined]
 [13] _collect(::Array{Test.AbstractTestSet,1}, ::Base.Generator{Array{Test.AbstractTestSet,1},TestReports.var"#2#3"}, ::Base.EltypeUnknown, ::Base.HasShape{1}) at ./array.jl:699
 [14] collect_similar(::Array{Test.AbstractTestSet,1}, ::Base.Generator{Array{Test.AbstractTestSet,1},TestReports.var"#2#3"}) at ./array.jl:628
 [15] map(::Function, ::Array{Test.AbstractTestSet,1}) at ./abstractarray.jl:2162
 [16] report(::ReportingTestSet) at /Users/malmiller/repos/TestReports.jl/src/to_xml.jl:77
 [17] top-level scope at REPL[29]:1
 [18] include_string(::Function, ::Module, ::String, ::String) at ./loading.jl:1088
```

--- a/src/to_xml.jl
+++ b/src/to_xml.jl
@@ -2,9 +2,7 @@
 # https://www.ibm.com/support/knowledgecenter/en/SSQ2R2_9.5.0/com.ibm.rsar.analysis.codereview.cobol.doc/topics/cac_useresults_junit.html
 # http://help.catchsoftware.com/display/ET/JUnit+Format
 
-function set_attribute!(node, attr, val)
-    link!(node, AttributeNode(attr, string(val)))
-end
+set_attribute!(node, attr, val) = setindex!(node, string(val), attr)
 
 function testsuites_xml(name, id, ntests, nfails, nerrors, x_children)
     x_testsuite = ElementNode("testsuites")
@@ -117,7 +115,8 @@ function to_xml(res::Pass)
 end
 
 function to_xml(v::Fail)
-    x_failure = failure_xml(string(v.data), string(v.test_type), string(v))
+    data = v.data === nothing ? "" : v.data  # Needed for V1.0
+    x_failure = failure_xml(string(data), string(v.test_type), string(v))
     x_testcase = testcase_xml(v, [x_failure])
     x_testcase, 1, 1, 0
 end

--- a/test/runtests.jl
+++ b/test/runtests.jl
@@ -30,6 +30,12 @@ end
     @test_reference "references/complexexample.txt" read(`$(Base.julia_cmd()) $(@__DIR__)/example.jl`, String) |> clean_report
 end
 
+@testset "Chained failing test - Issue #25" begin
+    ts = @testset ReportingTestSet begin
+            @test 1==1 && 1==0
+    end
+    @test report(ts) isa Any  # Would fail before #25
+end
 
 @testset "any_problems" begin
     @test any_problems(Pass(Symbol(), nothing, nothing, nothing)) == false

